bots/notify/src/main/java/org/openjdk/skara/bots/notify/prbranch/PullRequestBranchNotifier.java

```diff
@@ -50,6 +52,10 @@ private void pushBranch(PullRequest pr) {
             var seedRepo = hostedRepositoryPool.seedRepository(pr.repository(), false);
             seedRepo.fetch(pr.repository().url(), pr.headHash().hex());
             String branch = PreIntegrations.preIntegrateBranch(pr);
+            if (protectBranches) {
+                log.info("Protecting branch " + branch);
+                pr.repository().protectBranchPattern(branch);
+            }
             log.info("Creating new pull request pre-integration branch " + branch);
             seedRepo.push(pr.headHash(), pr.repository().url(), branch, true);
         } catch (IOException e) {
```

Comment:
I am wondering in the previous pushBranch, we didn't manually make the branch protected, but the branch pushed are protected, is this due to settings in GitLab?

Reply:
Before this change, we relied on a static configuration for the repository with certain branch patterns being protected. When this change goes live, we will need to adjust those patterns so that they do not apply to pr/X branches. Not sure if this answers your question.

Reply:
> Before this change, we relied on a static configuration for the repository with certain branch patterns being protected. When this change goes live, we will need to adjust those patterns so that they do not apply to pr/X branches. Not sure if this answers your question.

Yes, it answered my question. Thank you!